File: src/main/java/upbrella/be/user/repository/UserRepository.java
package upbrella.be.user.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import upbrella.be.user.entity.User;

import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {

    Optional<User> findBySocialId(Long socialId);

    boolean existsBySocialId(Long socialId);
}

File: src/main/kotlin/upbrella/be/user/repository/UserRepository.kt
package upbrella.be.user.repository

import org.springframework.data.jpa.repository.JpaRepository
import upbrella.be.user.entity.User
import java.util.Optional

interface UserRepository : JpaRepository<User, Long> {

    fun findBySocialId(socialId: Long): Optional<User>
    fun existsBySocialId(socialId: Long): Boolean
}